app/index.tsx

```diff
@@ -81,57 +72,54 @@ const parseDeepLinking = (url: string) => {
 	return null;
 };
 
-export default class Root extends React.Component<{}, IState> {
-	private listenerTimeout!: any;
+const Root = () => {
+	const { width: widthWindow, height: heightWindow, scale: scaleWindow, fontScale: fontScaleWindow } = Dimensions.get('window');
+	const [theme, setTheme] = useState(getTheme(initialTheme()));
+	const [themePreferences, setThemePreferences] = useState(initialTheme());
+	const [width, setWidth] = useState(widthWindow);
+	const [height, setHeight] = useState(heightWindow);
+	const [scale, setScale] = useState(scaleWindow);
+	const [fontScale, setFontScale] = useState(fontScaleWindow);
 
-	private onKeyCommands: any;
+	const listenerTimeout = useRef<any>();
+	const onKeyCommands = useRef<any>();
 
-	constructor(props: any) {
-		super(props);
-		this.init();
+	useEffect(() => {
+		init();
 		if (!isFDroidBuild) {
-			this.initCrashReport();
+			initCrashReport();
 		}
-		const { width, height, scale, fontScale } = Dimensions.get('window');
 		const theme = initialTheme();
-		this.state = {
-			theme: getTheme(theme),
-			themePreferences: theme,
-			width,
-			height,
-			scale,
-			fontScale
-		};
 		if (isTablet) {
-			this.initTablet();
+			initTablet();
 		}
 		setNativeTheme(theme);
-	}
+	}, []);
 
-	componentDidMount() {
-		this.listenerTimeout = setTimeout(() => {
+	useEffect(() => {
```

Comment:
Why do you need 2 useEffects with the same dependencies?
Are they going to run in parallel or something? 🤔